packages/extension-ui/src/Popup/Signing/LedgerSign.tsx

```diff
@@ -18,15 +22,35 @@ interface Props {
   className?: string;
   error: string | null;
   genesisHash?: string;
-  onSignature?: ({ signature }: { signature: HexString }) => void;
-  payload?: ExtrinsicPayload;
+  onSignature?: ({ signature }: { signature: HexString }, signedTransaction: HexString) => void;
+  payload?: SignerPayloadJSON;
   setError: (value: string | null) => void;
 }
 
+function getMetadataProof (chain: Chain, payload: SignerPayloadJSON) {
+  const m = chain.definition.rawMetadata || '0x00';
```

Comment:
I think we should rework this little portion, when no metadata is present there should be some sort of error that is thrown, and logged on the screen.